src/main/java/Search/Searcher.java

```diff
@@ -0,0 +1,142 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.TreeMap;
+
+import com.google.re2j.Matcher;
+import com.google.re2j.Pattern;
+
+public class Searcher {
+    private static final String DEFAULT_CHARSET = null; // Indicates platform default for FileUtils
+    private static final int PERIOD_MATCHES_NEWLINES_AND_SPACES = Pattern.DOTALL;
+    private static final String MATCH_INDICATOR = "||";
+    private static final char DELIMITER = ' ';
+
+    public Printable getPrintableSearchResult(Searchable searchable, String search) {
+        String transcript = getTranscript(searchable);
+        String timestamps = getTimestamps(searchable.timestampFile);
+        List<IntegerPair> matches = getMatches(transcript, search);
+        TreeMap<Integer, IntegerPair> transcriptTimestampIndices = mapTranscriptToTimestamps(transcript, timestamps);
+
+        return new Printable(matches, transcript, timestamps, searchable.filename, transcriptTimestampIndices);
+    }
+
+    private TreeMap<Integer, IntegerPair> mapTranscriptToTimestamps(String transcript, String timestamps) {
+        TreeMap<Integer, IntegerPair> transcriptTimestampIndices = new TreeMap<>();
+
+        int timestampStart = 0;
+        int timestampEnd = getNextDelimiterIndex(timestampStart, timestamps);
+
+        // base case for the first transcript word, which does not have a prior space
+        IntegerPair firstTimestamp = new IntegerPair(timestampStart, timestampEnd);
+        transcriptTimestampIndices.put(0, firstTimestamp);
+
+        for (int transcriptIndex = 0; transcriptIndex < transcript.length(); ++transcriptIndex) {
+            if (transcript.charAt(transcriptIndex) == DELIMITER) {
+                timestampStart = timestampEnd + 1;
+                timestampEnd = getNextDelimiterIndex(timestampStart, timestamps);
+                IntegerPair timestamp = new IntegerPair(timestampStart, timestampEnd);
+
+                if (timestampStart < timestampEnd) {
+                    transcriptTimestampIndices.put(transcriptIndex, timestamp);
+                }
+            }
+        }
+        return transcriptTimestampIndices;
+    }
+
+    int getNextDelimiterIndex(int current, String string) {
+        int nextDelimiter = string.indexOf(DELIMITER, current + 1);
+        if (nextDelimiter > current && nextDelimiter < string.length()) {
+            return nextDelimiter;
+        } else {
+            return string.length() - 1;
+        }
+    }
+
+    private String getTranscript(Searchable searchable) {
+        String transcript = getFileToString(searchable.transcriptFile);
+        return stripNewlines(transcript);
+    }
+
+    private String getTimestamps(File timestampFile) {
+        String timestamps = getFileToString(timestampFile);
+        return stripNewlines(timestamps);
+    }
+
+    private List<IntegerPair> getMatches(String transcript, String search) {
```

Comment:
semantic, but maybe findMatches instead of getMatches? Get implies that this function only gets values that are already in the transcript or search, but in reality it actually has to find them in the transcript string.

Reply:
renamed